### PR TITLE
Add NSArray support to MixpanelType

### PR DESCRIPTION
**Pull request by katherinepeterson**
Added NSArray type to MixpanelType. Needed this type while working with deserialized JSON

### CODE REVIEW

**Comment by zihejia** [comment]
hi @katherinepeterson , thank you so much! The file has been moved, could you do a rebase and solve the conflict then I will merge your PR.

**Comment by katherinepeterson** [comment]
@zihejia thanks, it's been resolved!

File: Sources/MixpanelType.swift [hunk 1]
```diff
@@ -210,6 +210,43 @@ extension Array: MixpanelType {
...
+        for element in self {
+            guard let _ = element as? MixpanelType else {
+                return false
+            }
+        }
+        return true
+    }
+
+    public func equals(rhs: MixpanelType) -> Bool {
+        if rhs is [MixpanelType] {
```
Comment (palayemre) [inline Sources/MixpanelType.swift:227]: I have 2 questions here;
- Why did not you choose to use `if let rhs = rhs as? [MixPanelType]` conditional binding?
- There is no check for `isValidNestedTypeAndValue` rhs? Can this cause a problem

Comment (katherinepeterson) [reply Sources/MixpanelType.swift:227]: @palayemre I followed the format of the other types in this file

**Review by zihejia — APPROVED** [review]
Thank you for providing the PR

**Pull request merged**